modules/openthread/platform/diag.c

```diff
@@ -68,3 +71,153 @@ void otPlatDiagAlarmCallback(otInstance *aInstance)
 {
 	ARG_UNUSED(aInstance);
 }
+
+/*
+ * To enable gpio diag commands, in Devicetree create `openthread` node in `/options/` path
+ * with `compatible = "openthread,config"` property and `diag-gpios` property,
+ * which should contain array of GPIO pin's configuration properties containing controller phandles,
+ * pin numbers and pin flags. e.g:
+ *
+ * options {
+ *		openthread {
+ *			compatible = "openthread,config";
+ *			diag-gpios = <&gpio0 0 GPIO_ACTIVE_HIGH>,
+ *				     <&gpio1 0 GPIO_ACTIVE_LOW>;
+ *		};
+ *	};
+ *
+ * To enable reading current gpio pin mode, define
+ * `CONFIG_GPIO_GET_DIRECTION` in prj.conf.
+ *
+ * Note: `<gpio>` in `diag gpio` commands is an index of diag-gpios array. For example shown above,
+ * `ot diag gpio mode 0` will return current mode of pin nmb 0 controlled by `gpio0` controller.
+ */
+#if DT_NODE_EXISTS(DT_PATH(options, openthread))
```

Comment:
I guess this would be the default path to put the properties, but I think the `compatible` would be preferred to match the node?